zproject/email_backends.py

```diff
@@ -108,3 +110,46 @@ def send_messages(self, email_messages: Sequence[EmailMessage]) -> int:
                 email_log_url = settings.ROOT_DOMAIN_URI + "/emails"
                 logging.info("Emails sent in development are available at %s", email_log_url)
         return num_sent
+
+
+class PersistentSMTPEmailBackend(EmailBackend):
+    def _open(self, **kwargs: Any) -> bool | None:
+        is_opened = super().open()
+        if is_opened:
+            self.opened_at = timezone_now()
+            return True
+
+        return is_opened
+
+    @override
+    def open(self, **kwargs: Any) -> bool | None:
+        is_opened = self._open()
+        if is_opened:
+            return True
+
+        status = None
+        time_elapsed = (timezone_now() - self.opened_at).seconds / 60
+        if settings.EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES != 0 and (
+            settings.EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES is None
+            or time_elapsed <= settings.EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES
+        ):
+            # No-op to ensure that we don't return a connection that has been
+            # closed by the mail server.
+            try:
+                assert self.connection is not None
+                status = self.connection.noop()[0]
+            except Exception:
+                pass
+
+        # In case of EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES = 0, we
+        # will always return a newly opened connection. The default
+        # smtp.py implementation of send_messages closes the connection
+        # after sending the emails if the connection was a new
+        # connection i.e. is_opened is True.
+
+        if status is None or status != 250:
+            # Close and connect again.
+            self.close()
+            is_opened = self._open()
+
+        return is_opened
```

Comment:
Per the above, I suspect this always needs to be False here -- or we need to be overriding the behaviour of `send_messages()` or `close()` so that a reconnection doesn't trigger the "oh, OK, I should close this, too" behaviour from the superclass.